dojo_plugin/pages/course.py

```diff
@@ -24,6 +25,17 @@ def get_letter_grade(dojo, grade):
             return letter_grade
     return "?"
 
+def clamp_ec(limit):
+    global_limit = [limit]
```

Comment:
Am I confused? Why is this a list?

Reply:
This is some insane python thing with the decorator instance holding the limiting value across functions.  If it is not a list the value isn't captured in the decorator.

Reply:
This is how scoping works inside python, what you're looking for is the `nonlocal` keyword.

I believe this is what you are trying to express:

```py
def clamp_ec(limit):
    def decorator(func):
        def wrapper(*args, **kwargs):
            nonlocal limit
            result = func(*args, **kwargs)
            clamped_result = min(result, limit)
            limit -= clamped_result
            return clamped_result
        return wrapper
    return decorator
```